### PR TITLE
24765 add support for enum props

### LINKED ISSUE #24765: Datasource property setters that take Enums as parameters are not handled properly
**Describe the bug**  
Attempting to set a datasource property whose setter method takes an Enum as a parameter causes an exception.  For example, setting the PostgreSQL specific data source property "autosave=always" (https://jdbc.postgresql.org/documentation/use/ ) doesn't work.
```
<server>
    <library id="POSTGRES_PROVIDER" name="POSTGRES_PROVIDER">
        <file name="${server.config.dir}/lib/postgresql-42.5.1.jar"/>
    </library>
    <jdbcDriver id="XA_POSTGRES_JDBC_Provider" javax.sql.DataSource="org.postgresql.xa.PGXADataSource" libraryRef="POSTGRES_PROVIDER">
    </jdbcDriver>
    <dataSource id="Data Source" jdbcDriverRef="XA_POSTGRES_JDBC_Provider" jndiName="enroleDataSource" >
    <properties.postgresql databaseName="${name}" password="${password}" portNumber="${port}" serverName="${host}" user="${user}" autosave="always" />
    </dataSource>
</server>
```
With this setting Liberty throws this error:
```
[WARNING ] DSRA8021W: Warning: error setting 'autosave'=always: java.lang.NoSuchMethodException: org.postgresql.jdbc.AutoSave.<init>(java.lang.String)
	at java.lang.Class.newNoSuchMethodException(Class.java:565)
	at java.lang.Class.getConstructor(Class.java:608)
	at [com.ibm.ws](http://com.ibm.ws/).jdbc.internal.JDBCDriverService.setProperty(JDBCDriverService.java:1005)
```
**Steps to Reproduce**  
Start server with the above datasource config.

**Expected behavior**  
The DS config code should properly convert property value strings into Enums.

**Diagnostic information:**  
 - OpenLiberty Version: affects all releases
 - Affected feature(s) jdbc-4.x
 - Java Version: N/A
 - server.xml configuration: see above
 - If it would be useful, upload the messages.log file found in `$WLP_OUTPUT_DIR/messages.log`

**Additional context**  
When trying to call the setter for a property that takes an Enum as the param, the current code attempts to construct an Enum via the ctor which is not possible.  Code needs to detect that the param type is an Enum and handle it properly.

### PR DESCRIPTION
**Pull request by frowe**
more updates

### CODE REVIEW

**Comment by frowe** [comment]
#build

**Comment by KyleAure** [comment]
@jantley-ibm @posmith I added the ID and L2 labels that were on the previous PR.  However, there is more messaging in this PR than in the last one.  You may need to re-review. Sorry for the confusion.

File: dev/com.ibm.ws.jdbc.metatype/resources/OSGI-INF/metatype/metatype.xml [hunk 1]
```diff
@@ -1397,6 +1397,12 @@
   <AD id="tcpKeepAlive"                  ibmui:group="Advanced" required="false" type="Boolean" name="%tcpKeepAlive" description="%tcpKeepAlive.desc"/>
   <AD id="user"                          ibmui:group="Advanced" required="false" type="String"  name="%user" description="%user.desc"/>
   <AD id="URL"                           ibmui:group="Advanced" required="false" type="String"  name="%URL" description="%URL.desc"/>
+  <AD id="autosave"                      ibmui:group="Advanced" required="false" type="String"                         name="%autosave"             description="autosave.desc">
+   <Option value="never"        label="never"/>
+   <Option value="always"       label="always"/>
+   <Option value="conservative" label="conservative"/>
+   <Option value="bogus"        label="bogus"/>
```
Comment (KyleAure) [inline dev/com.ibm.ws.jdbc.metatype/resources/OSGI-INF/metatype/metatype.xml]: Remove this bogus option (looks like it was only for testing)

**Review by KyleAure — CHANGES_REQUESTED** [review]

**Review by KyleAure — DISMISSED** [review]

**Comment by posmith** [comment]
L2 message review completed

**Comment by jantley-ibm** [comment]
@KyleAure LGTM

**Review by KyleAure — DISMISSED** [review]

**Comment by frowe** [comment]
#build

**Review by KyleAure — APPROVED** [review]

**Comment by frowe** [comment]
#libby

**Pull request merged**